File: 2026/07-July/11-Saturday/solution.py
from typing import List

class Solution:
    def countCompleteComponents(self, n: int, edges: List[List[int]]) -> int:
        graph = [[] for _ in range(n)]
        for u, v in edges:
            graph[u].append(v)
            graph[v].append(u)

        visited = [False] * n
        answer = 0

        for start in range(n):
            if visited[start]:
                continue

            stack = [start]
            visited[start] = True
            nodes = 0
            degree_sum = 0

            while stack:
                node = stack.pop()
                nodes += 1
                degree_sum += len(graph[node])

                for nxt in graph[node]:
                    if not visited[nxt]:
                        visited[nxt] = True
                        stack.append(nxt)

            if degree_sum // 2 == nodes * (nodes - 1) // 2:
                answer += 1

        return answer

File: 2026/07-July/11-Saturday/solution.java
import java.util.*;

class Solution {
    public int countCompleteComponents(int n, int[][] edges) {
        List<Integer>[] graph = new ArrayList[n];
        for (int i = 0; i < n; i++) graph[i] = new ArrayList<>();

        for (int[] e : edges) {
            graph[e[0]].add(e[1]);
            graph[e[1]].add(e[0]);
        }

        boolean[] visited = new boolean[n];
        int answer = 0;

        for (int i = 0; i < n; i++) {
            if (visited[i]) continue;

            Stack<Integer> stack = new Stack<>();
            stack.push(i);
            visited[i] = true;

            int nodes = 0;
            int degreeSum = 0;

            while (!stack.isEmpty()) {
                int cur = stack.pop();
                nodes++;
                degreeSum += graph[cur].size();

                for (int next : graph[cur]) {
                    if (!visited[next]) {
                        visited[next] = true;
                        stack.push(next);
                    }
                }
            }

            if (degreeSum / 2 == nodes * (nodes - 1) / 2)
                answer++;
        }

        return answer;
    }
}
